File: api/review.py
from flask import Flask, request, jsonify
from datetime import datetime

app = Flask(__name__)

def handler(request):
    """Vercel函数入口点"""
    if request.method == 'OPTIONS':
        return jsonify({'status': 'ok'}), 200, {
            'Access-Control-Allow-Origin': '*',
            'Access-Control-Allow-Headers': 'Content-Type,Authorization',
            'Access-Control-Allow-Methods': 'GET,PUT,POST,DELETE,OPTIONS'
        }
    
    if request.method != 'POST':
        return jsonify({'error': '只支持POST请求'}), 405, {
            'Access-Control-Allow-Origin': '*'
        }
    
    try:
        # 获取请求数据
        data = request.get_json() or {}
        review_date = data.get('date', datetime.now().strftime('%Y-%m-%d'))
        
        # 生成模拟复盘报告
        review_content = f"""# 股票交易复盘报告 - {review_date}

## 📊 市场环境分析
今日市场整体呈震荡上涨态势，成交量较昨日放大。

## 💼 个人交易回顾
- 交易笔数：3笔
- 交易金额：¥85,000
- 实现盈亏：+¥1,250
- 胜率：66.7%

## 📈 持仓分析
当前持仓5只股票，总市值¥485,000，浮盈¥15,000。

## 🎯 经验总结
1. 严格执行止损策略
2. 及时获利了结
3. 控制仓位风险

## 📅 明日计划
1. 关注新能源板块机会
2. 控制单日风险敞口
3. 保持理性交易心态

---
报告生成时间：{datetime.now().strftime('%Y-%m-%d %H:%M:%S')}
"""
        
        response_data = {
            'success': True,
            'content': review_content,
            'date': review_date,
            'generated_at': datetime.now().isoformat()
        }
        
        return jsonify(response_data), 200, {
            'Access-Control-Allow-Origin': '*'
        }
        
    except Exception as e:
        return jsonify({'error': f'生成复盘失败: {str(e)}'}), 500, {
            'Access-Control-Allow-Origin': '*'
        }

File: api/review.js
export default async function handler(req, res) {
  // 设置CORS头
  res.setHeader('Access-Control-Allow-Origin', '*');
  res.setHeader('Access-Control-Allow-Methods', 'GET, POST, PUT, DELETE, OPTIONS');
  res.setHeader('Access-Control-Allow-Headers', 'Content-Type, Authorization');

  if (req.method === 'OPTIONS') {
    res.status(200).end();
    return;
  }

  if (req.method !== 'POST') {
    res.status(405).json({ error: '只支持POST请求' });
    return;
  }

  try {
    const { date } = req.body || {};
    const reviewDate = date || new Date().toISOString().split('T')[0];

    // 模拟复盘报告
    const reviewContent = `# 交易复盘报告 - ${reviewDate}

## 一、市场整体环境复盘

### 指数与量能分析
今日市场整体呈震荡态势。

### 主要指数表现：
- 上证指数：需关注日均线趋势
- 深证成指：关注量能配合情况

## 二、个人交易复盘

### 今日交易概况
- 交易笔数：3笔
- 交易金额：¥50,000
- 实现盈亏：+¥1,200

### 交易明细分析
1. **买入 平安银行(000001)**
   - 买入价格：¥10.50
   - 买入数量：1000股
   - 买入理由：技术面突破，基本面良好

## 三、经验总结

### 做得好的地方
1. 严格执行止损策略
2. 及时获利了结

### 需要改进的地方
1. 仓位管理需要更加精细
2. 情绪控制有待提高

## 四、明日计划
1. 关注市场热点轮动
2. 控制仓位在合理范围内
`;

    res.status(200).json({
      success: true,
      content: reviewContent
    });
  } catch (error) {
    console.error('复盘错误:', error);
    res.status(500).json({ 
      error: '生成复盘失败: ' + error.message 
    });
  }
}